polkadot/runtime/common/src/slots/mod.rs

```diff
@@ -221,6 +241,51 @@ pub mod pallet {
 			};
 			Ok(())
 		}
+
+		/// Try to refund the lease deposit before the actual end of the lease.
+		///
+		/// This is only allowed if lease is ending within [`Config::EarliestRefundPeriod`] and the
+		/// parachain has no new lease periods coming up. This is useful for parachains who want to
+		/// get access to their funds they used in the last lease and rebid using same for the
+		/// next lease.
+		///
+		/// Can only be called by the parachain manager.
+		#[pallet::call_index(3)]
+		#[pallet::weight(T::WeightInfo::early_lease_refund())]
+		pub fn early_lease_refund(origin: OriginFor<T>, para: ParaId) -> DispatchResult {
+			// ensure caller is ForceOrigin or the manager of the parachain.
+			ensure_signed(origin.clone())
+				.map_err(|e| e.into())
+				.and_then(|who| -> DispatchResult {
+					ensure!(Some(who) == T::Registrar::manager_of(para), Error::<T>::NoPermission);
+					Ok(())
+				})
+				.or_else(|_| -> DispatchResult {
+					T::ForceOrigin::ensure_origin(origin)
+						.map(|_| ())
+						.map_err(|_| Error::<T>::NoPermission.into())
+				})?;
+
+			// check if lease is ending soon.
+			ensure!(
+				Self::lease_period_ending_soon(frame_system::Pallet::<T>::block_number()),
+				Error::<T>::PreconditionNotMet
+			);
+
+			// allow this iff parachain has one lease period left.
+			let leases = Leases::<T>::get(para);
+			ensure!(leases.len() == 1, Error::<T>::PreconditionNotMet);
```

Comment:
Add an integrity test to `assert!(T::EarliestRefundPeriod::get() <= T::LeasePeriod::get())`? Otherwise this could be misconfigured to let people refund more than one LP before expiry but they would still run into this error.